File: app/src/main/java/com/dimiaug/frustration/FrustrationApp.java
package com.dimiaug.frustration;

import android.app.Application;

import com.dimiaug.frustration.data.AppDatabase;
import com.dimiaug.frustration.repo.AppRepository;

public class FrustrationApp extends Application {

    public static final String TAG = "Frustration";

    @Override
    public void onCreate() {
        super.onCreate();
    }

    public AppRepository getRepository() {
        return AppRepository.getInstance(getDatabase());
    }

    private AppDatabase getDatabase() {
        return AppDatabase.getInstance(this);
    }

}

File: app/src/main/java/com/dimiaug/frustration/FrustrationApp.kt
package com.dimiaug.frustration

import android.app.Application
import com.dimiaug.frustration.common.di.commonModule
import com.dimiaug.frustration.features.play.di.playModule
import com.dimiaug.frustration.features.welcome.di.welcomeModule
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.GlobalContext.startKoin

class FrustrationApp : Application() {

    override fun onCreate() {
        super.onCreate()

        startKoin {
            androidLogger()
            androidContext(this@FrustrationApp)
            modules(
                commonModule,
                welcomeModule,
                playModule
            )
        }
    }

}